mdoc/src/main/scala/mdoc/internal/worksheets/WorksheetProvider.scala

```diff
@@ -35,26 +35,54 @@ class WorksheetProvider(settings: Settings) {
     val compiler = ctx.compiler(instrumented)
     val rendered =
       MarkdownBuilder.buildDocument(compiler, reporter, sectionInputs, instrumented, input.path)
-    val decorations = for {
+    val diagnostics = reporter.diagnostics.map(d => d: i.Diagnostic).toSeq
+
+    val evaluatedStatements = (for {
       section <- rendered.sections.iterator
       statement <- section.section.statements
-    } yield renderDecoration(statement)
+    } yield renderDecoration(statement)).toList
+
+    val exportableEvaluation =
+      diagnostics.filter(_.severity() == i.DiagnosticSeverity.Error) match {
```

Comment:
Maybe:
```scala
if (reporter.diagnostics.exists(_.severity() == i.DiagnosticSeverity.Error)){
  ju.Optional.empty[String]
} else {
  ju.Optional.of(zipStatements(input, evaluatedStatements))
}

```

Not sure why `reporter.diagnostics.map(d => d: i.Diagnostic).toSeq` is needed, could we just uses reported.diagnostics here ?

Reply:
ah, `exists` of course. 👍 

Regarding the `map(d => d: i.Diagnostic)` it's actually needed. Without it you'll end up with:

```
[bloop] [E] type mismatch;
 found   : java.util.List[mdoc.internal.io.Diagnostic]
 required: java.util.List[mdoc.interfaces.Diagnostic]
Note: mdoc.internal.io.Diagnostic <: mdoc.interfaces.Diagnostic, but Java-defined trait List is invariant in type E.
You may wish to investigate a wildcard type such as `_ <: mdoc.interfaces.Diagnostic`. (SLS 3.2.10)
```

Reply:
Could we have a pattern match or cast in exist then? With exist it should not be needed to iterate over full diags ever.

Reply:
I can try to play around a bit with it. Keep in mind this was also how it was before, I just moved it.

Reply:
Don't bother with it too much, I was just wondering if we can make it a bit more concise. Not a big deal.